assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../api';
+import Context from './context';
+
+function MediaProvider( { children } ) {
+	const { actions: { getMedia } } = useAPI();
+	const [ media, setMedia ] = useState( [] );
+	const [ mediaType, setMediaType ] = useState( '' );
```

Comment:
Media type filters and search terms seem to be more of a properties of the media panel and not the media provider itself. Are we ok these exposed globally?

Reply:
Maybe they'd be relevant to set for deep-linking? But other than that, I don't see any other components outside of the library using these.

Reply:
The reason I moved these here was becuase, imagine the following use case. 
- User filters media library to show videos with the word google. 
- User uploads image via drag and drop. 
- The media library should reset to default view, showing all video / images. 

This is a UX question.

Reply:
Yeah, we can approach it this way. Alternatively we could simply propagate a single "reset" signal, e.g. a pair of `actions: {resetMedia}` and `useMediaResetHandler( callback )`, to reset all dependent subcomponents. See what looks better to you for now. Separating the local state from library component might be also inconvenient. I think the question is do we ever need to set the filter to any value, or only resetting it.

Reply:
I think all state should be managed here, so having that as part of the library may make it hard to lose.

Reply:
@dvoytenko Maybe we could break this into another ticket and discuss this there? I am aware, that I don't want to slow this PR down with a larger discussion. I believe we are agreed that moving this state makes sense here.

Reply:
As we have more refactors of this component coming (with infinite scroll), I think it's fine to leave it as is for now.

Reply:
Agree. Let's leave for now.